examples/tokenvm/config/config.go

```diff
@@ -5,159 +5,44 @@ package config
-	c := &Config{nodeID: nodeID}
-	c.setDefault()
-	if len(b) > 0 {
-		if err := json.Unmarshal(b, c); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal config %s: %w", string(b), err)
-		}
-		c.loaded = true
-	}
-
-	// Parse any exempt sponsors (usually used when a single account is
-	// broadcasting many txs at once)
-	c.parsedExemptSponsors = make([]codec.Address, len(c.MempoolExemptSponsors))
-	for i, sponsor := range c.MempoolExemptSponsors {
-		p, err := codec.ParseAddressBech32(consts.HRP, sponsor)
-		if err != nil {
-			return nil, err
-		}
-		c.parsedExemptSponsors[i] = p
-	}
-	return c, nil
-}
-
-func (c *Config) setDefault() {
-	c.Config = vm.NewConfig()
-	c.LogLevel = logging.Info
+func New(b []byte) (*Config, error) {
 	gcfg := gossiper.DefaultProposerConfig()
-	c.GossipMaxSize = gcfg.GossipMaxSize
-	c.GossipProposerDiff = gcfg.GossipProposerDiff
-	c.GossipProposerDepth = gcfg.GossipProposerDepth
-	c.NoGossipBuilderDiff = gcfg.NoGossipBuilderDiff
-	c.VerifyTimeout = gcfg.VerifyTimeout
-	c.AuthVerificationCores = c.Config.AuthVerificationCores
-	c.RootGenerationCores = c.Config.RootGenerationCores
-	c.TransactionExecutionCores = c.Config.TransactionExecutionCores
-	c.StateFetchConcurrency = c.Config.StateFetchConcurrency
-	c.MempoolSize = c.Config.MempoolSize
-	c.MempoolSponsorSize = c.Config.MempoolSponsorSize
-	c.StateSyncServerDelay = c.Config.StateSyncServerDelay
-	c.StreamingBacklogSize = c.Config.StreamingBacklogSize
-	c.VerifyAuth = c.Config.VerifyAuth
-	c.StoreTransactions = defaultStoreTransactions
-	c.MaxOrdersPerPair = defaultMaxOrdersPerPair
-}
-
-func (c *Config) GetLogLevel() logging.Level                { return c.LogLevel }
-func (c *Config) GetTestMode() bool                         { return c.TestMode }
-func (c *Config) GetAuthVerificationCores() int             { return c.AuthVerificationCores }
-func (c *Config) GetRootGenerationCores() int               { return c.RootGenerationCores }
-func (c *Config) GetTransactionExecutionCores() int         { return c.TransactionExecutionCores }
-func (c *Config) GetStateFetchConcurrency() int             { return c.StateFetchConcurrency }
-func (c *Config) GetMempoolSize() int                       { return c.MempoolSize }
-func (c *Config) GetMempoolSponsorSize() int                { return c.MempoolSponsorSize }
-func (c *Config) GetMempoolExemptSponsors() []codec.Address { return c.parsedExemptSponsors }
-func (c *Config) GetTraceConfig() *trace.Config {
-	return &trace.Config{
-		Enabled:         c.TraceEnabled,
-		TraceSampleRate: c.TraceSampleRate,
-		AppName:         consts.Name,
-		Agent:           c.nodeID.String(),
-		Version:         version.Version.String(),
-	}
-}
-func (c *Config) GetStateSyncServerDelay() time.Duration { return c.StateSyncServerDelay }
-func (c *Config) GetStreamingBacklogSize() int           { return c.StreamingBacklogSize }
-func (c *Config) GetContinuousProfilerConfig() *profiler.Config {
```

Comment:
Some of these config values weren't even used... the profiler is interesting though so we could keep this if we wanted to provide it out-of-the-box (although it isn't used right now).